worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala

```diff
@@ -1330,21 +1445,28 @@ class PushDataHandler(val workerSource: WorkerSource) extends BaseMessageHandler
     batchOffsets match {
       case Some(batchOffsets) =>
         var index = 0
+        var indexOfSkipIndexes = 0
+        val skipListLength = sortedSkipIndexes.length
         var fileWriter: PartitionDataWriter = null
         while (index < fileWriters.length) {
-          fileWriter = fileWriters(index)
-          if (!writePromise.isCompleted) {
-            val offset = body.readerIndex() + batchOffsets(index)
-            val length =
-              if (index == fileWriters.length - 1) {
-                body.readableBytes() - batchOffsets(index)
-              } else {
-                batchOffsets(index + 1) - batchOffsets(index)
-              }
-            val batchBody = body.slice(offset, length)
-            writeData(fileWriter, batchBody, shuffleKey)
+          if (indexOfSkipIndexes < skipListLength && index == sortedSkipIndexes(
+              indexOfSkipIndexes)) {
+            indexOfSkipIndexes += 1
           } else {
-            fileWriter.decrementPendingWrites()
+            fileWriter = fileWriters(index)
+            if (!writePromise.isCompleted) {
+              val offset = body.readerIndex() + batchOffsets(index)
+              val length =
+                if (index == fileWriters.length - 1) {
+                  body.readableBytes() - batchOffsets(index)
+                } else {
+                  batchOffsets(index + 1) - batchOffsets(index)
+                }
+              val batchBody = body.slice(offset, length)
+              writeData(fileWriter, batchBody, shuffleKey)
```

Comment:
writeData may also encounter AlreadyClosedException

Reply:
If writeData encounter AlreadyClosedException, writePromise will invoke failure(new CelebornIOException(cause)) and pushMergedDataCallback will invoke onFailure.

Reply:
> If writeData encounter AlreadyClosedException, writePromise will invoke failure(new CelebornIOException(cause)) and pushMergedDataCallback will invoke onFailure.

yes, and I think better to handle this case as we already handle fileWriter closed before.

Reply:
This PR is intended to optimize process logic for pushMergedData and possibly create another PR for fileWriters with failures.

Reply:
I have supported some fileWriter failures by changing writePromise to Promise[Array[StatusCode]]. Thanks

Reply:
:clap::clap::clap: